packages/core/src/connectors/google/index.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * The Implementation of OpenID Connect of Google Identity Platform.
+ * https://developers.google.com/identity/protocols/oauth2/openid-connect
+ */
+
+import got, { RequestError as GotRequestError } from 'got';
+import { stringify } from 'query-string';
+import { z } from 'zod';
+
+import { connectorRequestTimeout } from '../constant';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorMetadata,
+  ConnectorType,
+  GetAccessToken,
+  GetAuthorizationUri,
+  GetUserInfo,
+  ValidateConfig,
+} from '../types';
+import { getConnectorConfig } from '../utilities';
+import { accessTokenEndpoint, authorizationEndpoint, scope, userInfoEndpoint } from './constant';
+
+export const metadata: ConnectorMetadata = {
+  id: 'google',
+  type: ConnectorType.Social,
+  name: {
+    en: 'Sign In with Google',
+    zh_CN: 'Google登录',
+  },
+  logo: './logo.png',
+  description: {
+    en: 'Sign In with Google',
+    zh_CN: 'Google登录',
+  },
+};
+
+const googleConfigGuard = z.object({
+  clientId: z.string(),
+  clientSecret: z.string(),
+});
+
+type GoogleConfig = z.infer<typeof googleConfigGuard>;
+
+export const validateConfig: ValidateConfig = async (config: unknown) => {
+  const result = googleConfigGuard.safeParse(config);
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error.message);
+  }
+};
+
+export const getAuthorizationUri: GetAuthorizationUri = async (redirectUri, state) => {
+  const config = await getConnectorConfig<GoogleConfig>(metadata.id);
+
+  return `${authorizationEndpoint}?${stringify({
```

Comment:
It's optional.
We use `state` and this request is performed by our backend service (not user client), so it can prevent replay protection some how.
<img width="886" alt="image" src="https://user-images.githubusercontent.com/10806653/156314529-7ac9ca5f-66e9-46c9-aece-b88f42cb70a3.png">